packages/datacommons-mcp/datacommons_mcp/services.py

```diff
@@ -157,75 +182,98 @@ async def search_topics_and_variables(
     if place2_dcid:
         base_place_dcids.append(place2_dcid)
 
-    search_tasks.append((query, base_place_dcids))
-
-    # The following queries are not needed for bilateral relationships where we append the place name(s) to the query.
+    search_tasks.append(SearchTask(query=query, place_dcids=base_place_dcids))
 
     # Place1 query: search for query + place1_name, filter by place2_dcid
     if place1_dcid:
         place1_place_dcids = [place2_dcid] if place2_dcid else []
-        search_tasks.append((f"{query} {place1_name}", place1_place_dcids))
+        search_tasks.append(SearchTask(query=f"{query} {place1_name}", place_dcids=place1_place_dcids))
```

Comment:
nit: the name `place1_place_dcids` is a little confusing considering it holds place2. I think line 189 and 190 can be combined. Similar suggestion for lines 194 and 195

```suggestion
        search_tasks.append(SearchTask(query=f"{query} {place1_name}", place_dcids=[place2_dcid] if place2_dcid else []))
```